src/server/db_slice.cc

```diff
@@ -442,12 +450,11 @@ pair<PrimeIterator, ExpireIterator> DbSlice::FindExt(const Context& cntx, string
         }
       };
 
-      //
       db.prime.CVCUponBump(change_cb_.back().first, res.first, bump_cb);
     }
-
-    res.first = db.prime.BumpUp(res.first, PrimeBumpPolicy{});
+    res.first = db.prime.BumpUp(res.first, PrimeBumpPolicy{bump_items_});
     ++events_.bumpups;
+    bump_items_.insert(key);
```

Comment:
I feel uncomfortable that you add string_view to the set since it's not clear here what's the life cycle of the `key` object outside of this function. I suggest that you actually store PrimaryKey. it is possible to use them as references.  another benefit of that - you won't need calling GetSlice in l CanBumpDown and you will be able to check it directly.

Reply:
check out `AsRef()` function in compact_object. Also, you will need to take care of key deletions to make it consistent.